Extend/Controller/Admin/PaymentMain.php

```diff
@@ -88,8 +90,86 @@ private function _isSavePossible()
         $sCountryCode = $aParams['oxpayments__wdoxidee_countrycode'];
         $bCountryCodeValid = $aParams['oxpayments__oxid'] !== SofortPaymentMethod::getName(true)
             || (preg_match('/^[a-z]{2}_[a-z]{2}$/', $sCountryCode) === 1);
+        $sCreditorId = $aParams['oxpayments__wdoxidee_creditorid'];
+        $bCreditorIdValid = $aParams['oxpayments__oxid'] !== SepaDirectDebitPaymentMethod::getName(true)
+            || $this->_creditorIdValidation($sCreditorId);
+
+        return $this->_isInputValid($bCredentialsValid, $bCountryCodeValid, $bCreditorIdValid);
+    }
+
+    /**
+     * Checks if it is possible to save the config
+     *
+     * @param bool $bCredentialsValid
+     * @param bool $bCountryCodeValid
+     * @param bool $bCreditorIdValid
+     *
+     * @return bool
+     *
+     * @since 1.0.1
+     */
+    private function _isInputValid($bCredentialsValid, $bCountryCodeValid, $bCreditorIdValid)
```

Comment:
I personally would expect a validation method to **validate the input** directly instead of other validation results.